kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessageFetchContext.java

```diff
@@ -313,9 +316,41 @@ private void readMessagesInternal(KafkaHeaderAndRequest fetch,
                             } else if (apiVersion <= 3) {
                                 magic = RecordBatch.MAGIC_VALUE_V1;
                             }
+
+                            String clientHost = fetch.getClientHost();
+                            // get group name from zk
+                            if (!requestHandler.currentConnectedGroup.containsKey(clientHost)) {
+                                String zkSubPath = clientHost.split(":")[0]
+                                        + "-" + fetch.getHeader().clientId();
+                                String groupId = KoPZKUtils.getData(requestHandler.pulsarService.getZkClient(),
+                                        "/consumer_client_group", zkSubPath);
+                                requestHandler.currentConnectedGroup.put(clientHost, groupId);
+                                log.info("get group name from zk for current connection:{} groupId:{}",
+                                        clientHost, groupId);
+                            }
+
+                            Consumer consumer = null;
+                            try {
+                                String groupId = requestHandler.currentConnectedGroup.get(clientHost);
+                                // create broker consumer
+                                requestHandler.getGroupCoordinator()
+                                        .getoffsetAcker().getConsumer(groupId, kafkaPartition).get();
+                                PersistentTopic persistentTopic = requestHandler.getTopicManager()
+                                        .getTopic(kafkaPartition.toString()).get();
+                                // get the broker consumer size
+                                log.info("consumer size " + persistentTopic.getSubscription(groupId)
+                                        .getConsumers().size());
```

Comment:
I'm confused about why `getConsumers` is null.

Reply:
I think it's because when a `PersistentTopic` is created, the ledger has no cursor, see https://github.com/apache/pulsar/blob/ee4cddf35970f6230b8c0a3f3c247e10e85b1973/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L240, where `ledger.getCursors()` returns empty.